.github/workflows/sims.yml

```diff
@@ -0,0 +1,119 @@
+name: Sims
+# Sims workflow runs multiple types of simulations (nondeterminism, import-export, after-import)
+# This workflow will run on all Pull Requests, if a .go, .mod or .sum file have been changed
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, 'skip-sims')"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.18
+      - name: Display go version
+        run: go version
+      - run: make build
+
+  install-runsim:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.18
+      - name: Display go version
+        run: go version
+      - name: Install runsim
+        run: export GO111MODULE="on" && go install github.com/cosmos/tools/cmd/runsim@v1.0.0
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/bin
+          key: ${{ runner.os }}-go-runsim-binary
+
+  test-sim-nondeterminism:
+    runs-on: ubuntu-latest
+    needs: [build, install-runsim]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.18
+      - name: Display go version
+        run: go version
+      - uses: technote-space/get-diff-action@v6
+        with:
+          PATTERNS: |
+            **/**.go
+            go.mod
+            go.sum
```

Comment:
Would it be possible to move this to the `build` job? If so this step could be removed from this and the other jobs to have a simpler config and also to stop the build job if there are no relevant files modified to make the workflow "faster".

Reply:
What do you mean by `build` job? We can join with the `test.yml`, but it will be extensive; the `sim.yml` is a huge file. Also, the tests here are running quickly, this will not take too much time, and I think it is more organized